save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DiktatDemoService.kt

```diff
@@ -1,63 +1,39 @@
 package com.saveourtool.save.demo.service
 
+import com.saveourtool.save.demo.config.ConfigProperties
 import com.saveourtool.save.demo.diktat.DiktatDemoAdditionalParams
-import com.saveourtool.save.demo.diktat.DiktatDemoMode
 import com.saveourtool.save.demo.diktat.DiktatDemoResult
-import com.saveourtool.save.demo.diktat.DiktatDemoTool
 
-import com.saveourtool.save.demo.runners.DiktatCliRunner
+import com.saveourtool.save.demo.runners.cli.DiktatCliRunner
 import org.springframework.stereotype.Service
-
-import java.io.File
-import java.util.*
+import java.nio.file.Path
+import kotlin.io.path.div
 
 /**
  * Demo service implementation for ktlint-demo/diktat-demo
  */
 @Service
-class DiktatDemoService : AbstractDemoService<DiktatDemoAdditionalParams, DiktatDemoResult> {
-    private fun generateName() = UUID.randomUUID().toString()
-    private fun generateDemoFile(tempDirPath: String): File = File("$tempDirPath${File.separator}test.kt")
-    private fun generateDemoConfig(tempDirPath: String): File = File("$tempDirPath${File.separator}config")
+class DiktatDemoService(
+    private val diktatCliRunner: DiktatCliRunner,
+    configProperties: ConfigProperties,
+) : AbstractDemoService<DiktatDemoAdditionalParams, DiktatDemoResult>(diktatCliRunner) {
+    private val tmpDir = Path.of(configProperties.fileStorage.location) / "tmp"
```

Comment:
To make the code more "Java-agnostic", you can use the `kotlin.io.path.Path` factory method instead.